signers/signer-solana/src/utils/main.ts

```diff
@@ -57,6 +57,13 @@ export async function executeSolanaTransaction(
       );
       return signedTransaction.serialize();
     } catch (e: any) {
+      if (tx.from !== solanaProvider.publicKey?.toString()) {
+        throw new SignerError(
+          SignerErrorCode.SIGN_TX_ERROR,
+          ' The connected account does not match the required account. Please ensure that you are connected with the correct account and try again.'
```

Comment:
```suggestion
          `Your connected account ${tx.from} doesn't match with the required account ${solanaProvider.publicKey?.toString()}. Please ensure that you are connected with the correct account and try again.'
```

Reply:
Isn't better to add this condition before signing the transaction?

Reply:
`${solanaProvider.publicKey?.toString()}` is null in this case and it is recommended to avoid displaying 'null' in error messages for better user experience and clarity. Also, I was not sure if `solanaProvider.publicKey` can be null and still the signing of transaction can be successful. I moved it to the top before signing the tx.